app/sample_manifest_excel/sample_manifest_excel/worksheet/test_worksheet.rb

```diff
@@ -135,18 +135,17 @@ def record_plate_samples # rubocop:todo Metrics/AbcSize
           plate_id = cur_sm_sample_asset.asset.plate.id
 
           # Validation errors here indicates problems we WANT not problems we HAVE
-          dynamic_attributes[sheet_row][:sanger_plate_id] =
-            if cgap
-              if validation_errors.include?(:sample_plate_id_duplicates)
-                'CGAP-99999'
-              elsif validation_errors.include?(:sample_plate_id_unrecognised_foreign)
-                "INVALID-#{plate_id.to_s.upcase}#{(plate_id % 10).to_s.upcase}"
-              else
-                "CGAP-#{plate_id.to_s(16).upcase}#{(plate_id % 16).to_s(16).upcase}"
-              end
+          dynamic_attributes[sheet_row][:sanger_plate_id] = if cgap
```

Comment:
Acceptable in Ruby, but I am not at all a fan of the assignment via value returned from an `if`/`elsif`/`else` block.  Please don't try to change this, I'm just saying it's something to avoid if at all possible when writing new code.